plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -109,24 +111,16 @@ func (g *genericScheduler) selectHost(priorityList schedulerapi.HostPriorityList
 		return "", fmt.Errorf("empty priorityList")
 	}
 
+	sort.Sort(priorityList)
 	maxScore := priorityList[0].Score
-	// idx contains indices of elements with score == maxScore.
-	idx := []int{}
-
-	for i, entry := range priorityList {
-		if entry.Score > maxScore {
-			maxScore = entry.Score
-			idx = []int{i}
-		} else if entry.Score == maxScore {
-			idx = append(idx, i)
-		}
-	}
+	lastMaxScoreIndes := sort.Search(len(priorityList), func(i int) bool { return priorityList[i].Score < maxScore })
```

Comment:
to be clear, it's not "lastMaxScoreIndex", but "firstNonMaxScoreIndex"

we should be explicit about this